### ISSUE TITLE
Documentation of migrating from Migrating from existing 2.x version is confusing

### ISSUE THREAD

**Issue by jmsjr** [springfox/springfox#3412]
Please take the time to search the repository, if your question has already been asked or answered.

- [x] What version of the library are you using? Is it the latest version? 
3.0.0

What kind of issue is this?

 - [x] Question. Is this a question about how to do a certain thing? 
Documentation.  Yes .. . migrating from 2.x.x to 3.0.0

 - [ ] Bug report. If you’ve found a bug, spend the time to write a failing test. Bugs with tests or 
       steps to reproduce get fixed faster. Here’s an example: https://gist.github.com/swankjesse/6608b4713ad80988cdc9
    - [ ] spring xml/java config that is relevant
    - [ ] springfox specific configuration if it makes sense
    - [ ] include any output you've received; logs, json snippets etc.; and what the expected output should be
    - [ ] if you have a repo that demonstrates the issue for bonus points! See [this example](https://github.com/springfox/springfox/issues/494) 

 - [ ] Feature Request. Start by telling us what problem you’re trying to solve. Often a solution
       already exists! Don’t send pull requests to implement new features without first getting our
       support. Sometimes we leave features out on purpose to keep the project small.

Please do consider ***starring*** this repository if you do like this project and find it useful. 

The documentation at http://springfox.github.io/springfox/docs/current/#migrating-from-existing-2-x-version says:

> 2.1.3. Migrating from existing 2.x version
> Spring Boot Applications
> Remove library inclusions of earlier releases. Specifically remove springfox-swagger2 and springfox-swagger-ui inclusions.
> 
> Remove the @EnableSwagger2 annotations

..  yes the very examples in the next sections:

http://springfox.github.io/springfox/docs/current/#springfox-spring-mvc-and-spring-boot
http://springfox.github.io/springfox/docs/current/#configuration-explained

... USES EnableSwagger2 annotations

```
@SpringBootApplication
@EnableSwagger2 
@ComponentScan(basePackageClasses = {
    PetController.class
})
public class Swagger2SpringBoot {
```

**Comment by dilipkrish** [comment]
Agreed thank you for pointing that out. Will address that! In the mean while let me know you have issues with the migration

**Comment by mmadoo** [comment]
In 3.5. Springfox Swagger UI, the documentation says:

> The Swagger UI page should then be available at http://localhost:8080/swagger-ui.html

while it should be: The Swagger UI page should then be available at http://localhost:8080/swagger-ui

**Comment by abuechler** [comment]
Also the example projects are not aligned with the documentation, e.g. the spring boot example:

- [doesn't just use the springfox-boot-starter dependency](https://github.com/springfox/springfox-demos/blob/master/boot-swagger/build.gradle)
- Uses the @EnableSwagger2 annotation https://github.com/springfox/springfox-demos/blob/0c31da356e7463cd405652c7bde25c100a7efbf6/boot-swagger/src/main/java/springfoxdemo/boot/swagger/Application.java#L45
- Uses [SwaggerUiWebMvcConfigurer](https://github.com/springfox/springfox-demos/blob/master/boot-swagger/src/main/java/springfoxdemo/boot/swagger/SwaggerUiWebMvcConfigurer.java) although the documentation states that the UI is only disabled for **non** spring boot applications

**Comment by abuechler** [comment]
The documentation states in the upgrade instructions to remove the springfox-swagger-ui:

[2.1.3. Migrating from existing 2.x version](http://springfox.github.io/springfox/docs/current/#migrating-from-existing-2-x-version)

1. Remove library inclusions of earlier releases. **Specifically remove springfox-swagger2 and springfox-swagger-ui inclusions**.

But then in section [3.5. Springfox Swagger UI](http://springfox.github.io/springfox/docs/current/#springfox-swagger-ui)

To include it in a standard Spring Boot application you can add the dependency as follows:
`
dependencies {
compile 'io.springfox:springfox-swagger-ui:3.0.0'
}
`

**Comment by ivangfr** [comment]
Hi @dilipkrish First of all, thanks for providing and maintaining this library.

I am upgrading from `2.9.2` to `3.0.0` but I am having some problems. Is it possible to provide a migration guide?

Thanks!

---
**Update**

I followed the one provided [here](https://springfox.github.io/springfox/docs/current/#migrating-from-existing-2-x-version) and it worked for me.
